### PR TITLE
gpio: Add Documentation

### PR DESCRIPTION
**Pull request by bauer-alex-174**
Hi Rahix,

since I did not see this module documented besides its naming in the `CHANGELOG.md` I desided to quickly add Doctrings and a function of my own, that toggles a gpio pin between `HIGH` and `LOW`.

If you have any suggestions to improve this PR further please let me know.

Best Regards

bauer-alex-174

### CODE REVIEW

File: Documentation/contrib/gpio.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+.. py:module:: tbot_contrib.gpio
```
Comment (Rahix) [inline Documentation/contrib/gpio.rst:1]: You also need to include this file in the index of documentation:

```diff
diff --git a/Documentation/index.rst b/Documentation/index.rst
index 67c193c..392ec11 100644
--- a/Documentation/index.rst
+++ b/Documentation/index.rst
@@ -115,6 +115,7 @@ To show what tbot can help you with, here are a few simple example usecases:
    modules/tc
    contrib/connector
    contrib/gdb
+   contrib/gpio
    contrib/linux
    contrib/locking
    contrib/swupdate
```

File: tbot_contrib/gpio.py [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
 
 class Gpio:
     def __init__(self, host: linux.LinuxShell, gpio_number: int):
+        """Initialize GPIO
+
+        Raspberry Pi Model 2 and above use as gpio_number their BCM GPIO Numbers
+
+        Args:
+            host (linux.LinuxShell):    Linux Shell
+            gpio_number (int):          GPIO Number
```
Comment (Rahix) [inline tbot_contrib/gpio.py:13]: The format for arguments is different in the sphinx configuration we're using:

```suggestion
        :param linux.LinuxShell host: Linux Shell
        :param int gpio_number: GPIO Number
```

File: tbot_contrib/gpio.py [hunk 3]
```diff
@@ -24,20 +41,77 @@ def set_direction(self, direction: str) -> None:
...
+        """
         (self._gpio_path / "active_low").write_text("1" if value else "0")
 
     def get_active_low(self) -> bool:
+        """Get Active Low
+
+        Reads value and returns direction ``in`` or ``out``
+
+        Returns:
+            bool: Value
```
Comment (Rahix) [inline tbot_contrib/gpio.py:59]: And similarly, return values are documented like this:

```suggestion
        :rtype: bool
        :returns: Value
```

File: tbot_contrib/gpio.py [hunk 4]
```diff
@@ -24,20 +41,77 @@ def set_direction(self, direction: str) -> None:
...
+        Args:
+            delay (float): Delay in s
+
+        Raises:
+            Exception: Direction
+        """
+        self.set_value(True)
+        time.sleep(delay)
+        self.set_value(False)
+        time.sleep(delay)
```
Comment (Rahix) [inline tbot_contrib/gpio.py:117]: I don't think _toggle_ is the right name for this.  To toggle a pin usually refers to set it LOW when it was previously HIGH and vice versa.  Not set and reset it in a single operation.

I would suggest calling it `pulse` and then have parameters `on_time` and `off_time` where `off_time` defaults to `None` and `None` meaning "use on_time for off_time".  In code:

```python
from typing import Optional

def pulse(self, on_time: float, off_time: Optional[float] = None) -> None:
    if off_time is None:
        off_time = on_time

    self.set_value(True)
    time.sleep(on_time)
    self.set_value(False)
    time.sleep(off_time)
```

What do you think?

**Review by Rahix — CHANGES_REQUESTED** [review]
Hey, thanks a lot for the contribution!  Please check my comments below.

File: tbot_contrib/gpio.py (hunk 4 above)
Comment (bauer-alex-174) [reply tbot_contrib/gpio.py:117]: Sounds good I'm going to change that

**Comment by bauer-alex-174** [comment]
I added your changes to the PR

**Review by Rahix — APPROVED** [review]
Looks good, thanks a lot!

**Pull request merged**